File: nan21.dnet.module.bd/nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/fin/ds/model/TaxLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.bd.fin.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeLov;

import net.nan21.dnet.module.bd.fin.domain.entity.Tax;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = Tax.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = TaxLovDs.fNAME) })
public class TaxLovDs extends AbstractTypeLov<Tax> {

    public static final String fRATE = "rate";
    public static final String fSUMMARY = "summary";

    @DsField()
    private Float rate;

    @DsField()
    private Boolean summary;

    public TaxLovDs() {
        super();
    }

    public TaxLovDs(Tax e) {
        super(e);
    }

    public Float getRate() {
        return this.rate;
    }

    public void setRate(Float rate) {
        this.rate = rate;
    }

    public Boolean getSummary() {
        return this.summary;
    }

    public void setSummary(Boolean summary) {
        this.summary = summary;
    }

}

File: nan21.dnet.module.bd/nan21.dnet.module.bd.ui.extjs/src/main/webapp/src/ds/TaxLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.bd.fin.ds.model.TaxLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"active", type:"boolean"},
		{name:"rate", type:"float", useNull:true},
		{name:"summary", type:"boolean"}
	]
});

Ext.define("net.nan21.dnet.module.bd.fin.ds.model.TaxLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"rate", type:"float", useNull:true},
		{name:"rate_From",type:"float", useNull:true},
		{name:"rate_To",type:"float", useNull:true},
		{name:"summary", type:"boolean", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.bd.fin.ds.param.TaxLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 